Composer/packages/client/src/store/reducer/index.ts

```diff
@@ -516,9 +525,16 @@ const publishFailure: (title: string) => ReducerFunc = (title) => (state, { erro
 const getPublishStatus: ReducerFunc = (state, payload) => {
   // the action below only applies to when a bot is being started using the "start bot" button
   // a check should be added to this that ensures this ONLY applies to the "default" profile.
-  if (payload.target.name === 'default' && payload.endpointURL) {
-    state.botStatus = BotStatus.connected;
-    state.botEndpoints[state.projectId] = `${payload.endpointURL}/api/messages`;
+  if (payload.target.name === 'default') {
+    if (payload.status == 200 && payload.endpointURL) {
+      state.botEndpoints[state.projectId] = `${payload.endpointURL}/api/messages`;
+      state.botStatus = BotStatus.connected;
+    } else if (payload.status == 202) {
+      state.botStatus = BotStatus.reloading;
+    } else if (payload.status == 500) {
+      state.botStatus = BotStatus.failed;
```

Comment:
Does the payload come with a message in addition to the status? If that is the case, that message could be valuable to surface or log for troubleshooting.

Reply:
yes, it is being passed into the botLoadErrorMsg one line below this line...